Service/ZdsToZgwService.php

```diff
@@ -0,0 +1,370 @@
+            $zaaktype = $this->entityManager->find('App:ObjectEntity', $zaaktypes[0]['_self']['id']);
+            $zaakArray['zaaktype'] = $zaaktype->getId()->toString();
+        } else {
+            $zaaktype = new ObjectEntity($zaakTypeEntity);
+            $zaaktype->hydrate($zaakArray['zaaktype']);
+
+            $this->entityManager->persist($zaaktype);
+            $this->entityManager->flush();
+
+            $zaakArray['zaaktype'] = $zaaktype->getId()->toString();
+        }//end if
+
+        $zaakArray = $this->connectEigenschappen($zaakArray, $zaaktype);
+        $zaakArray = $this->connectRolTypes($zaakArray, $zaaktype);
+
+        return $zaakArray;
+    }//end convertZaakType()
+
+    /**
+     * Receives a case and maps it to a ZGW case.
+     *
+     * @param array $data   The inbound data for the case
+     * @param array $config The configuration for the action
+     *
+     * @return array
+     */
+    public function zaakActionHandler(array $data, array $config): array
+    {
+        $this->configuration = $config;
+
+        $zaakEntity = $this->getEntity('https://vng.opencatalogi.nl/schemas/zrc.zaak.schema.json');
+
+        $mapping = $this->getMapping('https://opencatalogi.nl/schemas/zds.zdsZaakToZgwZaak.schema.json');
+
+        $zaakArray = $this->mappingService->mapping($mapping, $data['body']);
+
+        $zaakArray = $this->convertZaakType($zaakArray);
+
+        $zaken = $this->cacheService->searchObjects(null, ['identificatie' => $zaakArray['identificatie']], [$zaakEntity->getId()->toString()])['results'];
+        if (count($zaken) === 1) {
+            $zaak = $this->entityManager->find('App:ObjectEntity', $zaken[0]['_self']['id']);
+            $zaak->hydrate($zaakArray);
+
+            $this->entityManager->persist($zaak);
+            $this->entityManager->flush();
+
+            if ($mappingOut = $this->getMapping('https://opencatalogi.nl/schemas/zds.zgwZaakToBv03.schema.json')) {
+                $data['response'] = $this->createResponse($this->mappingService->mapping($mappingOut, $zaak->toArray()), 200);
+            }
+        } else if (count($zaken) > 1) {
+            $data['response'] = $this->createResponse(['Error' => 'More than one case exists with id '.$zaakArray['identificatie']]);
+        } else {
+            $data['response'] = $this->createResponse(['Error' => 'The case with id '.$zaakArray['identificatie'].' does not exist']);
+        }//end if
+
+        return $data;
+    }//end zaakActionHandler()
+
+    /**
+     * Receives a document and maps it to a ZGW EnkelvoudigInformatieObject.
+     *
+     * @param array $data   The inbound data for the case
+     * @param array $config The configuration for the action
+     *
+     * @return array
+     */
+    public function documentActionHandler(array $data, array $config): array
+    {
+        $this->configuration = $config;
+
+        $zaakEntity = $this->getEntity('https://vng.opencatalogi.nl/schemas/zrc.zaak.schema.json');
```

Comment:
This works fine,
Just feels like getting entities or other objects by reference could/should be also be in configuration, example:
https://github.com/CommonGateway/GeboorteVrijBRPBundle/blob/a36ee0e43e3dc9842feac28247913733714a1a38/Action/vrijbrp.ZgwToVrijbrpAction.action.json

https://github.com/CommonGateway/GeboorteVrijBRPBundle/blob/a36ee0e43e3dc9842feac28247913733714a1a38/Service/ZgwToVrijbrpService.php#L162